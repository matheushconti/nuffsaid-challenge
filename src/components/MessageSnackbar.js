import react from 'react'
import { Button, Snackbar, SnackbarContent } from '@material-ui/core'
import { messageTypes } from '../consts'

export default function MessageSnackbar(props){
  const { snackbar, handleCloseSnackbar } = props
  return (
    <Snackbar 
        open={snackbar.open} 
        key={snackbar.message} 
        onClose={handleCloseSnackbar} 
        autoHideDuration={2000} 
        anchorOrigin={{vertical: 'top',horizontal: 'right'}}
      >
        <SnackbarContent 
          style={{backgroundColor: messageTypes.error.color}}
          message= {snackbar.message}
          action={
            <Button onClick={() => handleCloseSnackbar()} color="inherit" size="small">
              X
            </Button>
          }
        />
      </Snackbar>
  )
}