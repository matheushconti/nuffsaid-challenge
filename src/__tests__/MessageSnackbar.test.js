import React from 'react'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import MessageSnackbar from '../components/MessageSnackbar'

configure({ adapter: new Adapter() });

const snackbarOpen = {open: true, message: "snackbar message"}
const snackbarClosed = {open: false, message: ""}
const handleCloseSnackbar = jest.fn()

describe('Testing MessageSnackbar', () => {
    test('rendering Snackbar open', () => {
        const tree = renderer.create(<MessageSnackbar snackbar={snackbarOpen} handleCloseSnackbar={handleCloseSnackbar} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('rendering Snackbar closed', () => {
        const tree = renderer.create(<MessageSnackbar snackbar={snackbarClosed} handleCloseSnackbar={handleCloseSnackbar} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
})