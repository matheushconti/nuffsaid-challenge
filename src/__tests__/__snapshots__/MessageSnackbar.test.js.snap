// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MessageSnackbar rendering Snackbar closed 1`] = `null`;
